### DISCUSSION TITLE
TextArea - onChange logic didn't work at first typing

### DISCUSSION THREAD

**Discussion by Kang-Eve** [ant-design/ant-design#42787] [Q&A]
antd version: ^5.5.1

In TextArea, I tried to check if the string of the field was valid through a regular expression.
`
        <TextArea
          value={props.value}
          defaultValue={props.defaultValue}
          disabled={props.disabled}
          onChange={(e) => {
            e.preventDefault()

            if (props.validChecker) {
              const valid = props.validChecker(e.target.value)
              if (!valid) {
                return
              }
            }

            props.onChange && props.onChange(e)
          }}
        />
`

Hand over validChecker to props, it is judged that it is not valid
When a string that does not fit the regular expression, it is return immediately, and I wanted to prevent it from being entered in the field.

This logic worked properly when the component was loaded and a string satisfying the regular expression was initially entered, but when a string violating the regular expression was initially entered, it did not work properly and a value was entered in the field.

How can I fix this logic? Is there a master who knows the solution?

**Comment by BoyYangzai** [comment]
@Kang-Eve 
hello bro 
can you take us a reproduction deom with online ide for quickly meeting error ? 
such as a `codesandbox`
